test/system/010-images.bats

```diff
@@ -359,4 +359,36 @@ EOF
     run_podman --root $imstore/root rmi --all
 }
 
+@test "podman images with concurrent removal" {
+    skip_if_remote "following test is not supported for remote clients"
+    local count=5
+
+    # First build $count images
+    for i in $(seq --format '%02g' 1 $count); do
+        cat >$PODMAN_TMPDIR/Containerfile <<EOF
+FROM $IMAGE
+RUN echo $i
+EOF
+        run_podman build -q -t i$i $PODMAN_TMPDIR
+    done
+
+    run_podman images
+    # Now remove all images in parallel and in the background and make sure
+    # that listing all images does not fail (see BZ 2216700).
+    for i in $(seq --format '%02g' 1 $count); do
+        echo "REMOVING IMAGE i$i"
+        timeout --foreground -v --kill=10 60 \
+                $PODMAN rmi i$i &
```

Comment:
At the very least we should store each pid then `wait` for it later and make sure all rmi's exited 0.

Reply:
That's done at the end of the test.